plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/connection/DBPConnectionConfiguration.java

```diff
@@ -107,6 +107,7 @@ public class DBPConnectionConfiguration implements DBPObject {
     private final List<DBWHandlerConfiguration> handlers;
     private final DBPConnectionBootstrap bootstrap;
     private DBPConnectionType connectionType;
+    private DBPDriverConfigurationType configurationType;
```

Comment:
It looks like this field is mostly `null`, it's better to add a default value and annotate get/set methods as `NotNull`